app/services/invoices/apply_provider_taxes_service.rb

```diff
@@ -116,7 +116,11 @@ def fees_amount_cents(tax)
     end
 
     def fetch_provider_taxes_result
-      taxes_result = Integrations::Aggregator::Taxes::Invoices::CreateService.call(invoice:)
+      taxes_result = if invoice.finalized?
```

Comment:
I think this does not work. In most cases we call `invoice.finalized!` after we call `Invoices::ComputeAmountsFromFees` (please take a look at `Invoices::CreatePayInAdvanceChargeService`). That being said, invoice will be in `generating` status and  `Integrations::Aggregator::Taxes::Invoices::CreateDraftService` will be called instead of `Integrations::Aggregator::Taxes::Invoices::CreateService`. IMO we should probably have one path just for draft invoices and other path for everything else